pkg/utils/cel/quantity_test.go

```diff
@@ -0,0 +1,297 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cel
+
+import (
+	"maps"
+	"math"
+	"testing"
+	"time"
+
+	"github.com/google/cel-go/common/types"
+	"github.com/google/cel-go/common/types/ref"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+var expectedTime = time.Date(2006, 1, 2, 15, 4, 5, 0, time.UTC)
+
+func TestQuantityCalculation(t *testing.T) {
+	tests := []struct {
+		name       string
+		expression string
+		expected   string
+	}{
+		{
+			name:       "parse tebibyte quantity",
+			expression: `Quantity("1Ti")`,
+			expected:   `1Ti`,
+		},
+		{
+			name:       "add tebibyte",
+			expression: `Quantity("1Ti") + Quantity("1Ti")`,
+			expected:   `2Ti`,
+		},
+		{
+			name:       "subtract gibibyte",
+			expression: `Quantity("1Ti") - Quantity("512Gi")`,
+			expected:   `512Gi`,
+		},
+		{
+			name:       "negate tebibyte",
+			expression: `-Quantity("1Ti")`,
+			expected:   `-1Ti`,
+		},
+		{
+			name:       "tebibyte multiply int",
+			expression: `Quantity("1Ti") * 2`,
+			expected:   `2Ti`,
+		},
+		{
+			name:       "tebibyte multiply uint",
+			expression: `Quantity("1Ti") * 2u`,
+			expected:   `2Ti`,
+		},
+		{
+			name:       "tebibyte multiply float",
+			expression: `Quantity("1Ti") * 0.5`,
+			expected:   `512Gi`,
+		},
+		{
+			name:       "tebibyte divide int",
+			expression: `Quantity("1Ti") / 2`,
+			expected:   `512Gi`,
+		},
+		{
+			name:       "tebibyte divide uint",
+			expression: `Quantity("1Ti") / 2u`,
+			expected:   `512Gi`,
+		},
+		{
+			name:       "tebibyte divide float",
+			expression: `Quantity("1Ti") / 2.0`,
+			expected:   `512Gi`,
+		},
+		{
+			name:       "parse millicpu quantity",
+			expression: `Quantity("1m")`,
+			expected:   `1m`,
+		},
+		{
+			name:       "millicpu multiply int",
+			expression: `Quantity("1m") * 2`,
+			expected:   `2m`,
+		},
+		{
+			name:       "millicpu multiply float",
+			expression: `Quantity("1m") * 1.0`,
+			expected:   `1m`,
+		},
+		{
+			name:       "millicpu floor to zero",
+			expression: `Quantity("1m") * 0.5`,
+			expected:   `0`,
+		},
+		{
+			name:       "millicpu divide int",
+			expression: `Quantity("5m") / 2`,
+			expected:   `2m`,
+		},
```

Comment:
At the moment the smallest CPU unit in the K8s is 1m, anything less than that is meaningless.